tests/subsys/input/input_shell/testcase.yaml

```diff
@@ -0,0 +1,14 @@
+# SPDX-License-Identifier: Apache-2.0
+
+tests:
+  input.input_shell:
+    tags: input
+    platform_allow:
+      - native_posix
+      - native_posix_64
+    harness: console
+    harness_config:
+      type: multi_line
+      regex:
+        - "I: input event: dev=NULL             SYN type= 1 code=  2 value=0"
+        - "I: input event: dev=NULL             SYN type= 1 code=  2 value=1"
```

Comment:
Have you considered using the ztest subsystem for this test instead of a `while(true)` loop and a regex parser?

Reply:
No but I did now, so I got it working with ZTEST, to do that I had to define a custom log output processor, intercept the messages, have the log subsystem process them, dump them into a buffer and then check the output. This is how it looks like: https://gist.github.com/fabiobaltieri/976c8829f0f70c35e2ae3ba24081f4b4.

Now that works fine but looking at it I feel like there's way too much logging internals for what is not a logging test, and doing that just to rip off the log messages when that's exactly what the twister harness does feels a bit silly and feels like it may break and get in the way as the logging subsystem evolves.

What do you think? I'd lay towards leaving it as a twister validated test.

Reply:
That's a fair point. I'm okay with keeping the validation in twister instead of ztest.